### ISSUE TITLE
@doc "Couldn't find the following modules: Stdlib"

### ISSUE THREAD

**Issue by vphantom** [ocaml/dune#5229]
## Expected Behavior

Building without errors. 😉

## Actual Behavior

Using `dune build @all @doc` with or without `--watch`, I almost consistently get errors like this:

```text
File "../../lib_lwt/.gxd_lwt.objs/byte/gxd_lwt.odoc":
Couldn't find the following modules:
  Lwt Lwt_result Stdlib
File "../../lib/.gxd.objs/byte/gxd.odoc":
Couldn't find the following modules:
  Stdlib
```

These libraries are installed, as the build itself works fine. Even the generated documentation seems okay.

## Reproduction

Clearly there's something about my setup that is unique enough to cause this rare issue, but I'm at a loss as to how to narrow things down.  I'm hoping you might suggest things to verify or attempt so I can get a bit closer to what you'd need to reproduce this.

## Specifications

- Version of `dune` (output of `dune --version`): 2.9.1
- Version of `ocaml` (output of `ocamlc --version`): 4.13.1 (built with Musl GCC 10.2.1)
- Operating system (distribution and version): Debian GNU/Linux 10 (buster), kernel 4.16.6 on x86_64

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag): https://gist.github.com/vphantom/9ccb68c832e39980ec0fa1afa302397a

**Comment by rgrinberg** [comment]
I suppose we aren't specifying dependencies on external libraries correctly. cc @jonludlam

**Comment by kdvolder** [comment]
> Clearly there's something about my setup that is unique enough to cause this rare issue

Whatever it is, I seem to have the same issue.I am getting very similar errors. All related to not able to find 'Stdlib'.

**Comment by jonludlam** [comment]
This is currently expected. What's happening is that odoc is unable to do anything with stuff that comes from Stdlib - e.g. it can't expand `Map`s or `Set`s, can't link to e.g. `Buffer.t` and so on. This is just because dune is currently only building the docs for your package, not any dependencies. This is something I'd like to see fixed in some way, but I haven't personally had the time to do it.

**Comment by Alizter** [comment]
Will you pull in the dependencies and generate the docs locally, kind of like the jsoo build rules?

**Comment by jonludlam** [comment]
Yep - either that or have something maintain a switch-wide set of docs that can be linked to, odig style.

**Comment by dybber** [comment]
Strange. The issue only occured the first time I executed `dune build @doc` on subsequent executions it worked fine.
